src/dither_api.jl

```diff
@@ -36,35 +41,57 @@ function dither!(
     return alg(out, img, args...; kwargs...)
 end
 
+# Before `__dither!` is called, additional checks and defaults are run for AbstractCustomColorDither...
+function _dither!(
+    out::GenericImage,
+    img::GenericImage,
+    alg::AbstractCustomColorDither,
+    cs::AbstractVector{<:Pixel},
+    args...;
+    metric::DifferenceMetric=DE_2000(),
+    kwargs...,
+)
+    length(cs) >= 2 ||
+        throw(DomainError(steps, "Color scheme for dither needs >= 2 colors."))
+    return __dither!(out, img, alg, cs, metric, args...; kwargs...)
```

Comment:
> It might also be worth considering that methods automatically use channel-wise dithering when calling dither on a color image without providing a color palette (instead of having to wrap an algorithm in SeparateSpace). Binary dithering would only be applied when calling dither on a Gray image.
I can't think of a simpler interface, but maybe this would hide too much functionality in the dispatch? What do you think?

Do you mean something like

```julia
    out_cv = channelview(out)
    img_cv = channelview(img)
    for c in axes(img_cv, 1)
        __dither!(view(out_cv, c, :, :), view(img_cv, c, :, :), alg, cs, metric, args...; kwargs...)
    end
    return out
```

Reply:
Yes, exactly! I'll make another PR for it with some more ideas for interface changes. :)